dags/iop/repository.py

```diff
@@ -0,0 +1,62 @@
+import os
+from io import BytesIO
+from typing import IO
+
+from common.repository import IRepository
+from common.s3_service import S3Service
+
+
+class IOPRepository(IRepository):
+    ZIPED_DIR: str = "raw/"
+    EXTRACTED_DIR: str = "extracted/"
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.s3 = S3Service(os.getenv("IOP_BUCKET_NAME", "iop"))
+
+    def get_all_raw_filenames(self):
+        return [
+            f.key.split("/")[-1]
+            for f in self.s3.objects.filter(Prefix=self.ZIPED_DIR).all()
+        ]
+
+    def find_all(self, filenames_to_process=None):
+        ret_dict = {}
+        filenames = []
+        filenames = (
+            filenames_to_process
+            if filenames_to_process
+            else self.__find_all_extracted_files()
+        )
+        for file in filenames:
+            file_parts = file.split("/")
+            last_part = file_parts[-1]
+            filename_without_extension = last_part.split(".")[0]
+            if filename_without_extension not in ret_dict.keys():
+                ret_dict[filename_without_extension] = dict()
+            ret_dict[filename_without_extension][
+                "xml" if self.is_meta(last_part) else "pdf"
```

Comment:
maybe extract this before, assign a meaningful variable name and then refer to this value, so 
```
extension = "xml" if self.is_meta(last_part) else "pdf"
ret_dict[filename_without_extension][extension] = file
```